packages/checkout/validation/test/reducers.js

```
@@ -0,0 +1,238 @@
+		expect( nextState ).toEqual( {
+			existingError: {
+				message: 'This is an existing error message',
+				hidden: false,
+			},
+			newError: {
+				message: 'This is a new error',
+				hidden: false,
+			},
+		} );
+	} );
+	//Checks if it clears all the errors
+	it( 'Clear all validation errors', () => {
+		const state = {
+			existingError: {
+				message: 'This is an existing error message',
+				hidden: false,
+			},
+		};
+		const clearAllErrors = {
+			type: types.CLEAR_ALL_VALIDATION_ERRORS,
+		};
+		const nextState = reducer( state, clearAllErrors );
+		expect( nextState ).toEqual( {} );
+	} );
+	//Checks if it clears a single error
+	it( 'Clear single validation error', () => {
+		const state = {
+			existingError: {
+				message: 'This is an existing error message',
+				hidden: false,
+			},
+			testError: {
+				message: 'This is error should not be removed',
+				hidden: false,
+			},
+		};
+		const clearError = {
+			type: types.CLEAR_VALIDATION_ERROR,
+			error: 'existingError',
+		};
+		const nextState = reducer( state, clearError );
+		expect( nextState ).not.toHaveProperty( 'existingError' );
+		expect( nextState ).toHaveProperty( 'testError' );
+	} );
+	//Checks if it hides an error
+	it( 'Hide single validation error', () => {
+		const state = {
+			existingError: {
+				message: 'This is an existing error message',
+				hidden: false,
+			},
+			testError: {
+				message: 'This is error should not be removed',
+				hidden: false,
+			},
+		};
+		const testAction = {
+			type: types.HIDE_VALIDATION_ERROR,
+			error: 'existingError',
+		};
+		const nextState = reducer( state, testAction );
+		expect( nextState ).toEqual( {
+			existingError: {
+				message: 'This is an existing error message',
+				hidden: true,
+			},
+			testError: {
+				message: 'This is error should not be removed',
+				hidden: false,
+			},
+		} );
+	} );
+	//Checks if it display a single error
+	it( 'Show single validation error', () => {
+		const state = {
```

It would be good to add 2 more errors to the state here that have `{ hidden: false }` to test wether this action will also hide any errors that are currently visible, and only show the specified one

---

That's not the goal of this action actually, it's just to change whether a specific action is hidden or not. But good idea to add a visible one and ensure it remains visible after this action is applied to a different error!